mdata/write_buffer_test.go

```diff
@@ -0,0 +1,357 @@
+package mdata
+
+import (
+	"fmt"
+	"testing"
+
+	"gopkg.in/raintank/schema.v1"
+)
+
+func testAddAndGet(t *testing.T, reorderWindow uint32, flush func(uint32, float64), testData, expectedData []schema.Point, expectAddFail bool) {
+	b := NewReorderBuffer(reorderWindow, 1, flush)
+	gotFailure := false
+	for _, point := range testData {
+		success := b.Add(point.Ts, point.Val)
+		if !success {
+			gotFailure = true
+		}
+	}
+	if expectAddFail && !gotFailure {
+		t.Fatal("Expected an add to fail, but they all succeeded")
+	}
+	returned := b.Get()
+
+	if len(expectedData) != len(returned) {
+		t.Fatal("Length of returned and testData data unequal", len(returned), len(expectedData))
+	}
+	if !pointSlicesAreEqual(expectedData, returned) {
+		t.Fatal(fmt.Sprintf("Returned data does not match expected data\n%+v\n %+v", testData, expectedData))
+	}
+}
+
+func pointSlicesAreEqual(a, b []schema.Point) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i := range a {
+		if (a[i].Ts != b[i].Ts) || (a[i].Val != b[i].Val) {
+			return false
+		}
+	}
+	return true
+}
+
+func unsort(data []schema.Point, unsortBy int) []schema.Point {
```

Comment:
can this be simplified? what is the goal exactly here? do we just want to shuffle the array? what is unsortBy for?